tests/test_betting_strategy.py

```diff
@@ -18,3 +29,32 @@ def test_answer_decision(
     betting_strategy = MaxAccuracyBettingStrategy()
     direction: bool = betting_strategy.calculate_direction(market_p_yes, estimate_p_yes)
     assert direction == expected_direction
+
+
+def test_rebalance() -> None:
+    strategy = MaxAccuracyBettingStrategy()
+
+    mock_amount = TokenAmount(amount=5, currency=Currency.xDai)
+    tiny_amount = TokenAmount(amount=0.0001, currency=Currency.xDai)
+    mock_existing_position = Position(
+        market_id="0x123",
+        amounts={
+            OmenAgentMarket.get_outcome_str_from_bool(True): mock_amount,
+            OmenAgentMarket.get_outcome_str_from_bool(False): mock_amount,
+        },
+    )
+    mock_answer = ProbabilisticAnswer(p_yes=Probability(0.9), confidence=0.5)
+    mock_market = Mock(OmenAgentMarket, wraps=OmenAgentMarket)
+    mock_market.get_tiny_bet_amount.return_value = tiny_amount
+    mock_market.current_p_yes = 0.5
+    mock_market.currency = Currency.xDai
+    mock_market.id = "0x123"
+
+    trades = strategy.calculate_trades(mock_existing_position, mock_answer, mock_market)
+    # assert 1 buy trade and 1 sell trade
+    buy_trade = next(iter([t for t in trades if t.trade_type == TradeType.BUY]))
+    assert buy_trade
```

Comment:
There would be an exception, if no buy trade is found, so this assert is unnecessary:

```
>>> next(iter([]))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
StopIteration
```

```suggestion
    assert buy_trade.trade_type == TradeType.BUY
```